django_select2/views.py

```diff
@@ -52,7 +52,12 @@ def get(self, request, *args, **kwargs):
 
     def get_queryset(self):
         """Get QuerySet from cached widget."""
-        return self.widget.filter_queryset(self.term, self.queryset)
+        kwargs = {
+            model_field_name: self.request.GET.get(form_field_name)
+            for form_field_name, model_field_name in self.widget.dependent_fields.items()
+            if form_field_name in self.request.GET
```

Comment:
This exception doesn't exist, but you can work around that or just raise a 404. Thats fine with me.